tests/Functional/Feature/FixedWindowRateLimiterFeatureTest.php

```diff
@@ -0,0 +1,282 @@
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Tests\Functional\Feature;
+
+use Doctrine\Common\Cache\CacheProvider;
+use JsonException;
+use Monolog\Logger;
+use OAT\SimpleRoster\Entity\LineItem;
+use OAT\SimpleRoster\Exception\DoctrineResultCacheImplementationNotFoundException;
+use OAT\SimpleRoster\Repository\LineItemRepository;
+use OAT\SimpleRoster\Tests\Traits\DatabaseTestingTrait;
+use OAT\SimpleRoster\Tests\Traits\LoggerTestingTrait;
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class FixedWindowRateLimiterFeatureTest extends WebTestCase
+{
+    use DatabaseTestingTrait;
+    use LoggerTestingTrait;
+
+    /** @var KernelBrowser */
+    private $kernelBrowser;
+
+    /** @var CacheProvider */
+    private $resultCacheImplementation;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->kernelBrowser = self::createClient();
+
+        $ormConfiguration = $this->getEntityManager()->getConfiguration();
+        $resultCacheImplementation = $ormConfiguration->getResultCacheImpl();
+
+        if (!$resultCacheImplementation instanceof CacheProvider) {
+            throw new DoctrineResultCacheImplementationNotFoundException(
+                'Doctrine result cache implementation is not configured.'
+            );
+        }
+
+        $this->resultCacheImplementation = $resultCacheImplementation;
+        $this->setUpDatabase();
+        $this->loadFixtureByFilename('userWithReadyAssignment.yml');
+
+        $this->setUpTestLogHandler('security');
+    }
+
+    /**
+     * @SuppressWarnings(PHPMD.Superglobals)
+     */
+    public function testItAcceptsRequestsAfterItervalOfTwoSeconds(): void
+    {
+        $_ENV['RATE_LIMITER_FIXED_WINDOW_ROUTES'] = 'updateLineItems';
+        $_ENV['RATE_LIMITER_FIXED_WINDOW_LIMIT'] = 2;
+        $_ENV['RATE_LIMITER_FIXED_WINDOW_INTERVAL'] = '2 seconds';
+
+        $this->executeUpdateLineItemsRequest();
+
+        self::assertSame(Response::HTTP_OK, $this->kernelBrowser->getResponse()->getStatusCode());
+
+        $this->executeUpdateLineItemsRequest();
+
+        self::assertSame(Response::HTTP_OK, $this->kernelBrowser->getResponse()->getStatusCode());
+
+        $this->resetKernel();
```

Comment:
> Why do we need to reset kernel between calls?

Since there is more than one http call, calls would be isolated. And for us it means, that logs would be unaccessible after 2nd request (though they do appear in log file).
So, there are two ways, either calling `self::ensureKernelShutdown()` or doing the trick as in [there](https://github.com/oat-sa/simple-roster/blob/develop/tests/Functional/Action/Lti/GetUserAssignmentLtiLinkActionTest.php#L456) (lines 456-458)
p.s.
Though I don't see a real need to do that in additional method

cc @luisfn ^^